File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        #return list(map(int, list(str(int(''.join(map(str, digits)))+1))))
        carry = 0
        for i in range(len(digits)-1, -1, -1):
            if digits[i] < 9:
                digits[i] += 1
                return digits
            else:
                digits[i] = 0
                carry = 1
        
        return digits if carry == 0 else [1] + digits

File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        for (int i = n - 1; i >= 0; i--) {
            int digit = digits[i] + 1;
            if (digit == 10) {
                digits[i] = 0;
            } else {
                digits[i]++;
                return digits;
            } 
        }
        digits.insert(digits.begin(), 1);
        return digits;
    }
};
//class Solution {
//public:
//    vector<int> plusOne(vector<int>& digits) {
//        int n = digits.size();
//        for (int i = n-1; i >= 0; i--) {
//            int digit = digits[i];
//            if (digit == 9) {
//                digits[i] = 0;
//            } else {
//                digits[i]++;
//                return digits;
//            } 
//        }
//        digits.insert(digits.begin(), 1);
//        return digits; 
//    }
//};